pkg/graveler/ref/manager.go

```diff
@@ -37,487 +37,134 @@ type CommitNode struct {
-	return err
-}
-
-func (m *Manager) DeleteTag(ctx context.Context, repositoryID graveler.RepositoryID, tagID graveler.TagID) error {
-	_, err := m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		r, err := tx.Exec(
-			`DELETE FROM graveler_tags WHERE repository_id = $1 AND id = $2`,
-			repositoryID, tagID)
-		if err != nil {
-			return nil, err
-		}
-		if r.RowsAffected() == 0 {
-			return nil, graveler.ErrNotFound
-		}
-		return nil, nil
-	})
-	if errors.Is(err, db.ErrNotFound) {
-		return graveler.ErrTagNotFound
+func (m *KVManager) CreateTag(ctx context.Context, repositoryID graveler.RepositoryID, tagID graveler.TagID, commitID graveler.CommitID) error {
+	t := &graveler.TagData{
+		Id:       tagID.String(),
+		CommitId: commitID.String(),
 	}
-	return err
-}
-
-func (m *Manager) ListTags(ctx context.Context, repositoryID graveler.RepositoryID) (graveler.TagIterator, error) {
-	_, err := m.GetRepository(ctx, repositoryID)
+	tagKey := graveler.TagPath(tagID)
+	err := m.kvStore.SetMsgIf(ctx, graveler.TagPartition(repositoryID), tagKey, t, nil)
 	if err != nil {
-		return nil, err
+		if errors.Is(err, kv.ErrPredicateFailed) {
+			err = graveler.ErrTagAlreadyExists
+		}
+		return err
 	}
-	return NewTagIterator(ctx, m.db, repositoryID, IteratorPrefetchSize), nil
+	return nil
 }
 
-func (m *Manager) GetCommitByPrefix(ctx context.Context, repositoryID graveler.RepositoryID, prefix graveler.CommitID) (*graveler.Commit, error) {
-	key := fmt.Sprintf("GetCommitByPrefix:%s:%s", repositoryID, prefix)
-
-	commit, err := m.batchExecutor.BatchFor(key, MaxBatchDelay, batch.BatchFn(func() (interface{}, error) {
-		return m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-			records := make([]*commitRecord, 0)
-			// LIMIT 2 is used to test if a truncated commit ID resolves to *one* commit.
-			// if we get 2 results that start with the truncated ID, that's enough to determine this prefix is not unique
-			err := tx.Select(&records, `
-					SELECT id, committer, message, creation_date, parents, meta_range_id, metadata, version, generation
-					FROM graveler_commits
-					WHERE repository_id = $1 AND id LIKE $2 || '%'
-					LIMIT 2`,
-				repositoryID, prefix)
-			if errors.Is(err, db.ErrNotFound) {
-				return nil, graveler.ErrNotFound
-			}
-			if err != nil {
-				return nil, err
-			}
-			if len(records) == 0 {
-				return "", graveler.ErrNotFound
-			}
-			if len(records) > 1 {
-				return "", graveler.ErrRefAmbiguous // more than 1 commit starts with the ID prefix
-			}
-			return records[0].toGravelerCommit(), nil
-		}, db.ReadOnly())
-	}))
-	if errors.Is(err, db.ErrNotFound) {
-		return nil, graveler.ErrCommitNotFound
-	}
-	if err != nil {
-		return nil, err
-	}
-	return commit.(*graveler.Commit), nil
+func (m *KVManager) DeleteTag(ctx context.Context, repositoryID graveler.RepositoryID, tagID graveler.TagID) error {
+	tagKey := graveler.TagPath(tagID)
+	return m.kvStore.DeleteMsg(ctx, graveler.TagPartition(repositoryID), tagKey)
```

Comment:
Is `ErrNotFound` expected here by the caller? Because `kvStore.DeleteMsg` won't give that to you :)